src/posix/Makefile.am

```diff
@@ -62,6 +62,7 @@ LDADD_COMMON                                                           = \
 
 if OPENTHREAD_TARGET_LINUX
 LDADD_COMMON                                                          += \
+    -lanl                                                                \
```

Comment:
what is lanl?

Reply:
Using `getaddrinfo_a()` requires to link wih `-lanl`. https://man7.org/linux/man-pages/man3/getaddrinfo_a.3.html